### PR TITLE
fix: Proxy empty password was authorized.

### PR DESCRIPTION
**Pull request by Matovidlo**
Jira: [PSGO-728](https://keboola.atlassian.net/browse/PSGO-728)

**Changes:**
- Add regression test and write down how the handler works with password, empty or filled, cookie empty or filled or corrupted.

-----------

[PSGO-728]: https://keboola.atlassian.net/browse/PSGO-728?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: internal/pkg/service/appsproxy/proxy/apphandler/authproxy/basicauth/handler.go [hunk 1]
```diff
@@ -74,6 +74,14 @@ func (h *Handler) CookieExpiration() time.Duration {
...
 }
 
+// ServeHTTPOrError serves basic authorization pages based on these conditions:
+// nothing || cookie == nil -> render Login page GET/POST (200), no error
+// form[password=""] -> Please enter a correct password (200), error
+// form[password="b"] -> Please enter a correct password (200), error
+// form[password="a"] -> Authorized, set cookie and redirect to URL of user. (301), no error
+// cookie != nil && not signout && authorized -> Go to upstream (Based on app state), no error
+// cookie != nil && signout -> unset cookie, redirect to Login pageWriter (303), go to _proxy/SignInPath, no error
+// cookie != nil && unauthorized -> Cookie has expired (200), error
```
Comment (Matovidlo) [inline internal/pkg/service/appsproxy/proxy/apphandler/authproxy/basicauth/handler.go:84]: I don't know whether this should be on top of module, to let someone know how the handler works(should work).

**Review by hosekpeter — APPROVED** [review]

File: internal/pkg/service/appsproxy/proxy/apphandler/authproxy/basicauth/handler.go [hunk 2]
```diff
@@ -146,11 +155,17 @@ func (h *Handler) ServeHTTPOrError(w http.ResponseWriter, req *http.Request) err
...
 
 func (h *Handler) isAuthorized(password string, cookie *http.Cookie) error {
-	if password != "" && !h.basicAuth.IsAuthorized(password) {
+	// When password was not provided and cookie is set to wrong password, return `Cookie has expired.`
+	if cookie != nil && password == "" {
+		return h.isCookieAuthorized(cookie)
+	}
+
+	// When no password provided, or wrong password is provided against configured one, return error
+	if password == "" || password != "" && !h.basicAuth.IsAuthorized(password) {
```
[the remark this answers is not in the document]
Comment (Matovidlo) [reply internal/pkg/service/appsproxy/proxy/apphandler/authproxy/basicauth/handler.go]: changed
